tests/conftest.py

```diff
@@ -2683,3 +2684,67 @@ def ceph_toolbox(request):
     if not (deploy or teardown or skip_ocs):
         # Creating toolbox pod
         setup_ceph_toolbox()
+
+
+@pytest.fixture(scope='function')
+def node_drain_teardown(request):
+    """
+    Tear down function after Node drain
+
+    """
+    def finalizer():
+        """
+        Make sure that all cluster's nodes are in 'Ready' state and if not,
+        change them back to 'Ready' state by marking them as schedulable
+
+        """
+        scheduling_disabled_nodes = [
+            n.name for n in get_node_objs() if n.ocp.get_resource_status(
+                n.name
+            ) == constants.NODE_READY_SCHEDULING_DISABLED
+        ]
+        if scheduling_disabled_nodes:
+            schedule_nodes(scheduling_disabled_nodes)
+
+        # Remove label created for DC app pods on all worker nodes
+        node_objs = get_node_objs()
+        for node_obj in node_objs:
+            if 'dc' in node_obj.get().get('metadata').get('labels').keys():
+                remove_label_from_worker_node(
+                    [node_obj.name], label_key="dc"
+                )
+
+    request.addfinalizer(finalizer)
+
+
+@pytest.fixture(scope='function')
+def node_restart_teardown(request, nodes):
+    """
+    Make sure all nodes are up again
+    Make sure that all cluster's nodes are in 'Ready' state and if not,
+    change them back to 'Ready' state by restarting the nodes
+
+    """
+    def finalizer():
+        # Start the powered off nodes
+        nodes.restart_nodes_by_stop_and_start_teardown()
+        try:
+            node.wait_for_nodes_status(status=constants.NODE_READY)
+        except ResourceWrongStatusException:
+            # Restart the nodes if in NotReady state
+            not_ready_nodes = [
+                n for n in node.get_node_objs() if n
+                .ocp.get_resource_status(n.name) == constants.NODE_NOT_READY
+            ]
+            if not_ready_nodes:
+                log.info(
+                    f"Nodes in NotReady status found: {[n.name for n in not_ready_nodes]}"
+                )
+                nodes.restart_nodes(not_ready_nodes)
+                node.wait_for_nodes_status(status=constants.NODE_READY)
+
+        # Check ceph health
+        assert ceph_health_check(tries=80), "Ceph cluster health is not OK"
+        log.info("Ceph cluster health is OK")
```

Comment:
Can the ceph health be handled separately? this seems out of scope to the node_restart_teardown().
WDYT?

Reply:
In my testing due to amount of data there will me be minimal rebalancing happening after node restart, i think it is better we validate health as well in node teardown.